kanidm_unix_int/src/daemon.rs

```diff
@@ -89,14 +90,15 @@ impl Decoder for TaskCodec {
     type Error = io::Error;
 
     fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
-        match serde_cbor::from_slice::<TaskResponse>(&src) {
-            Ok(msg) => {
+        let maybe_task_response = serde_cbor::from_slice::<TaskResponse>(&src)
```

Comment:
Do we need this let? Couldn't we just directly return after the ok()?

Reply:
I was trying to add clarity, but we could do it that way. I was also wondering if we could just do this:
```suggestion
        let maybe_task_response = serde_cbor::from_slice(&src); // type inferencer has enough information
        src.clear();
        Ok(maybe_task_response.ok())
```

I'm curious why we have so many nested `Result`'s and `Option`'s as return types, doesn't that somewhat defeat the point? Shouldn't we always just be returning a `Result`, and make more expressive `Err` types for better readability? I think the compiler would also be able to do more optimizations like this as well.